static/templyte/src/app/tutorial/tutorial.component.ts

```diff
@@ -0,0 +1,57 @@
+import {AuthService} from '../providers/auth.service';
+import {Router} from '@angular/router';
+import {User} from 'firebase';
+import {ApiInterfaceService} from '../providers/api-interface.service';
+import {Component, OnInit} from '@angular/core';
+
+import {UploadService, Upload} from '../upload/upload.service';
+import {AngularFireDatabase} from 'angularfire2/database';
+import {MatDialog} from '@angular/material/dialog';
+import {InputValidateDialogComponent} from '../input-validate-dialog/input-validate-dialog.component';
+import {UploadSuccessDialogComponent} from '../upload-success-dialog/upload-success-dialog.component';
+
+@Component({
+    selector: 'app-tutorial',
+    templateUrl: './tutorial.component.html',
+    styleUrls: ['./tutorial.component.css']
+})
+
+export class TutorialComponent implements OnInit {
+	user: User = null;
+
+    constructor(
+        private apiInterfaceService: ApiInterfaceService,
+        private authService: AuthService,
+        private db: AngularFireDatabase,
```

Comment:
Do we need this?

Reply:
we need, private authService: AuthService and private router: Router, removed the others.